### ISSUE TITLE
Overseerr loading times are excrutiatingly slow

### ISSUE THREAD

**Issue by abhineetgupta** [sct/overseerr#3510]
### Description

It takes more than a few minutes to load any page on Overseerr. The screen just stays in `Loading` mode. Even the Settings page which is text only can take multiple minutes to load.

### Version

1.33.0

### Steps to Reproduce

1. Go to http://OVER.SEE.RR.IP:PORT/requests
2. Wait for page to load
3. Wait some more
4. Refresh hoping page load will be faster
5. Repeat from step 2

### Screenshots

Requests page after waiting for 4 minutes<img width="1669" alt="Screen Shot 2023-06-18 at 10 21 27 AM" src="https://github.com/sct/overseerr/assets/15706629/870a5a51-d1c5-43ab-a70e-aeb781f461b5">

Settings page after waiting for 30 seconds<img width="1670" alt="Screen Shot 2023-06-18 at 10 21 48 AM" src="https://github.com/sct/overseerr/assets/15706629/96ff3b58-88ef-484d-9808-c3304cdd8fe5">

### Platform

desktop

### Device

Macbook Pro

### Operating System

MacOS 11.7.7

### Browser

Firefox 114.0

### Additional Context

Overseerr is running on Synology NAS DS918+ using `linuxserver.io` docker image.

### Code of Conduct

- [X] I agree to follow Overseerr's Code of Conduct

**Comment by mdezzi** [comment]
I am experiencing the same issue. Extremely long load times on every screen, it started after upgrading to 1.33. I am running hotio docker image. 

There doesnt appear to be anything in the logs that hints at why the loading is so slow.

**Comment by 338lm** [comment]
SO SLOW, it's painful. This started immediately after the "upgrade" to 1.33.0

**Comment by barkcollar** [comment]
Same! My load times have been painfully slow as well.

**Comment by Bovive** [comment]
I am also experiencing this. 1.33.0 snap version on Ubuntu.

**Comment by 338lm** [comment]
this resolved for me after a couple of days. I think it was an issue with TMDB at the time

**Comment by angelgtc87** [comment]
mine slow as of now, tried my friend overseer and is smooth :(

**Comment by rmatec** [comment]
Overseerr has been extremely slow since I started running it on my Synology DS920+ for some reason.
At first I thought the resources were not enough but in comparison I can run a full Gitlab instance way faster.
My internet is very fast so that is not the issue.
Could it be the linuxserver.io is slow? Has anybody tried a different image?

**Comment by T0uc0** [comment]
Mine is also very slow, the page refreseh very often when I search movies, I take more than 30sec for load the homepage, manage request is slow too... I tried to change DNS, move to host connection, recreate the container, update the container. Nothing work... Anyone have a solution?

**Comment by barkcollar** [comment]
My performance issues resolved when I moved from a reverse proxy to using a
Cloudflare tunnel for remote access.

**Comment by chrishuman0923** [comment]
Same here! Running on custom homelab build and it is SO slow to do anything.

I am running behind Traefik, so I am gonna take some time to test the solution that @barkcollar implemented (remove from reverse proxy) and see what happens.

Just odd that none of the app logs for any of my containers points to a reason this would be this slow.

**Comment by rxbrad** [comment]
Loading is extremely slow on a Cloudflare-tunneled Overseerr here, and it has been for months.  As everyone notes above, nothing in the logs seems to give a clue as to why.  I just spend upwards of 30 seconds staring at pulsing grey squares before it finally starts loading in art.  But randomly, about 10% of the time, everything loads in quickly.

![image](https://github.com/sct/overseerr/assets/13750413/e9131c4d-f7ed-4c4a-8292-1ccc0bf86f5e)

Loading the server using the non-tunneled local network IP is _usually_ (but not always) faster, so I suspect there's a magic caching setting somewhere, either on Cloudflare or in Overseerr that might help.

**Comment by c11umw** [comment]
Same issue here with linuxserver image of overseerr
Cleared logs and cache folders, forced container DNS to cloudflared and allowed docker to access all NAS CPUs
None of it helped.

Reverted to 1.32.5 and - it's back to loading fast.
Definitely something quirky with 1.33+

```
docker pull linuxserver/overseerr:1.32.5
```

**Comment by zelik888** [comment]
> Same issue here with linuxserver image of overseerr Cleared logs and cache folders, forced container DNS to cloudflared and allowed docker to access all NAS CPUs None of it helped.
> 
> Reverted to 1.32.5 and - it's back to loading fast. Definitely something quirky with 1.33+
> 
> ```
> docker pull linuxserver/overseerr:1.32.5
> ```

Reverting to 1.32.5 did not fix the issue for me.  Moving overseerr to SSD drive does fix the issue. Strange that this would require SSD to run smoothly.

**Comment by Schu-** [comment]
I also seem to be plagued by this bug. The image always has been on an SSD and no changes on my side except updates. I have tested behind a proxy and direct on the system but the bug occurs no matter where it is loaded from. 

Still poking around on my own for logs or something to point out what might be going on. But wanted to add to this bug report for information.

Thanks!

**Comment by Schu-** [comment]
Hello All,

I just wanted to comment on this since I was able to figure out and fix my issue. After a bit more scouring the web, I was able to find a reddit post talking about slowness. [https://www.reddit.com/r/Overseerr/comments/sirat6/discover_page_takes_3060_seconds_to_load/](https://www.reddit.com/r/Overseerr/comments/sirat6/discover_page_takes_3060_seconds_to_load/). This post talks about adding a DNS command to docker run `--dns 1.1.1.1`. I made this change on my system and Overseerr worked 100% immediately. This made me question my own DNS issue and did some digging. On my side I found a misconfigured ACL rule to allow my container to reach my DNS server. Once I fixed that all was good on my side.

I would recommend anyone having this issue to review the post about and possibly add that docker run command.

Thanks!

**Comment by rxbrad** [comment]
I've tried the DNS trick previously, and just gave it another attempt.

Still no help.

This issue is baffling & random, and unfortunately nothing in the logs is
of any help.

**Comment by mdezzi** [comment]
@Schu- thanks for the tip. I am using adguard for local DNS, so that could be a potential slowdown. I added the dns entry in the overseerr container and i'll monitor it to see if it is any faster.

**Comment by chrishuman0923** [comment]
I just added the extra docker run arg that @Schu- suggested. It immediately fixed my issue!

I am running my network DNS through PiHole. PiHole resolves requests using 1.1.1.1 already so maybe I have something misconfigured but not sure where to start looking. I'm just glad this has resolved it for the time being.

**Comment by drn** [comment]
FWIW — in my case, it turned out that setting DNS to `1.1.1.1` and `1.0.0.1` in my docker image was the culprit of the slowness. I had accepted that it wasn't "great" for a while — but then pulling up overseerr logs it was logging a number of `getaddrinfo EAI_AGAIN` errors.

I removed that config, blew away the old overseerr db/cache (kept the settings.json), and recreated the docker image and it's blazing fast now. I could've sworn I tested it without that DNS config, but perhaps that was on my old host (moved from a synology nas to a mac mini — but it was still slow and I gave up for a while).

Here's my current docker config:

```
  overseerr:
    image: linuxserver/overseerr:latest
    container_name: overseerr
    environment:
      - PUID=501
      - PGID=20
      - TZ=America/Los_Angeles
    volumes:
      - $HOME/docker/overseerr/config:/config
    restart: unless-stopped
    network_mode: bridge
    depends_on:
      - sonarr
      - radarr
    ports:
      - "5055:5055"
```

Hope this helps someone else! It's been bothering me nuts for months and it seems to be finally fast AF

**Comment by zelik888** [comment]
Overseerr works great on SSD. I had it on a docker in proxmox CT running on SSD and it was very fast. Moved it to synology docker on regular HDD and it slowed to a crawl. DNS fixes didn't help. Installed SSD cache on my NAS and it's zippy again.

Perhaps nuking the database and starting fresh would fix everything.

**Comment by c11umw** [comment]
> Perhaps nuking the database and starting fresh would fix everything.

I've tried a few things including the forced 1.1.1.1 DNS as well as allowing the container to use all cores (--cpuset-cpus="0-3"), but none of it seemed to help.
Tried a fresh data path/db and it is a lot smoother and quicker.
So much so that I forgot it's using the latest and not the older version pre-slowdown 

Steps to replicate:
Stop the container, remove images, create a new folder and copy the settings.json across to it.
Amended the data path in the container to the new location and started up.
After that I force ran the jobs to re-sync with plex library etc

The only gotcha in doing this is that it doesn't keep the request list, so you may have to make a note and then re-request the content and assign to the user that requested. If you forgot to do it then it's still available on the old data path to go back and take note.

But yea, it looks a lot better :)

Docker CLI for reference:
```
echo Starting overseerr
docker run -d \
--name=overseerr \
--network=proxylan \
-p 5055:5055 \
-e PUID=1000 \
-e PGID=1000 \
-e TZ=Europe/London \
-v /volume1/docker/overseerr2:/config \
--restart=unless-stopped \
linuxserver/overseerr:latest
```

**Comment by c11umw** [comment]
> But yea, it looks a lot better :)

I take it back.
1 reboot later and it's back to a screen full of grey and a spinning loading timer 🙄

**Comment by rxbrad** [comment]
None of the suggestions anywhere in these comments work...  If they do, it's always just a coincidence.  Randomly, maybe 10% of the time, Overseerr loads great.  The rest of the time, it doesn't.

Steps I've taken and can confirm DON'T work...
- Manually setting the Docker container to use Cloudflare DNS
- Manually setting the Docker container to use Google DNS
- Running Overseerr on an SSD
- Totally nuking Overseerr and starting from scratch
- Enabling image cache

The only problems that ever even show up in my browser console logs are random failures to download avatar images from Plex.

> GET https://plex.tv/users/xxxxxxx/avatar?c=xxxxxx

**Comment by viart** [comment]
- Synology ds1621+
- Docker (was) on HDD

Tried quite a bit to solve it, including debugging the app code in the container (+DNS/network) with no obvious issues on the app side...

Decided to add an nvme pool and move docker there, on the way noticed that here I have a checkmark here:
![Control Panel > Shared Folder > Docker [edit]](https://github.com/sct/overseerr/assets/488812/c9660c66-7a66-4d6d-8b7d-3f249328c23a)

*Unchecking it* before moving `docker` folder to the nvme pool solved the issue for me. 

I suspect the "data integrity" thing was the root - Overseer is pretty aggressive with file operations (cache) and Synology with counting checksums on it :)

**Comment by timdonovanuk** [comment]
There seems to be a dozen posts about this on reddit and clearly a ton of people are having this issues and found their way to this thread. I am also using Cloudflare to access, so it seems to be DNS related. But why? No other web app I'm hosting has this issue.

**Comment by rxbrad** [comment]
I recently put explicit network information on all of my Docker containers. Since then, my Overseerr has been running consistently zippy.

I wonder if this is a MTU or IPV6 issue?  Here's the network info I use for the container, in XML (I use Portainer stacks).

```
networks:
  default:
    enable_ipv6: true
    driver: bridge
    driver_opts:
      com.docker.network.enable_ipv6: "true"
      com.docker.network.driver.mtu: 1400
    ipam:
      driver: default
      config:
       - subnet: 192.168.140.0/24
         gateway: 192.168.140.1
       - subnet: fd00:0:140::/64
         gateway: fd00:0:140::1
```

**Comment by wiffleman** [comment]
Also struggling with this. Tried all of the aforementioned tips with the exception of migrating docker onto a NVME SSD. 
No errors seen on the logs either.

Just reinforcing the nature of this bug. Also running behind a reverse proxy (Traefik). All other containers are snappy and work well.

**Comment by timdonovanuk** [comment]
10 to 25 seconds to load most of the elements on the page:

![image](https://github.com/sct/overseerr/assets/8156439/7562a81f-e8ce-48d6-a06a-bed98ca7d18e)

**Comment by rxbrad** [comment]
Lowering the MTU to 1400 on the Docker container has resolved the slowness on my Overseerr install for about 2 months now.

I had the constant slowdowns until I specifically set the MTU. I have had zero slowdowns since.

**Comment by timdonovanuk** [comment]
I'm not even running in a docker container 😄 The MTU tweak seems to have worked, but it's really random, sometimes I think I've fixed it but the issue comes back. It's certainly not a lack of resources on my server.

**Comment by wiffleman** [comment]
> Lowering the MTU to 1400 on the Docker container has resolved the slowness on my Overseerr install for about 2 months now.
> 
> I had the constant slowdowns until I specifically set the MTU. I have had zero slowdowns since.

Curious how you did this? Wondering if you can let me know if there are docker-compose commands I can write to make this change?

I have also recently tried changing to the stock sctx/overseerr image as opposed to the linuxserver one. I'm not sure if it's just my conscience playing with me but it *seems* to load quite a bit faster... will contineu to monitor.

**Comment by rxbrad** [comment]
> > Lowering the MTU to 1400 on the Docker container has resolved the slowness on my Overseerr install for about 2 months now.
> > I had the constant slowdowns until I specifically set the MTU. I have had zero slowdowns since.
> 
> Curious how you did this? Wondering if you can let me know if there are docker-compose commands I can write to make this change?
> 
> I have also recently tried changing to the stock sctx/overseerr image as opposed to the linuxserver one. I'm not sure if it's just my conscience playing with me but it _seems_ to load quite a bit faster... will contineu to monitor.

Here's the network section of my XML (a few posts up from here)

https://github.com/sct/overseerr/issues/3510#issuecomment-1860896187

**Comment by tylerpace** [comment]
I haven't seen this raised yet, but how many people experiencing the slowness are running their Overseerr data directory from a NFS share? Overseerr uses SQLite to store internal data and SQLite is notoriously slow on NFS shares due to issues with how file locking works (or doesn't) over NFS. 

The [SQLite devs themselves](https://sqlite.org/faq.html#q5) warn against using SQLite on NFS.

> But use caution: this locking mechanism might not work correctly if the database file is kept on an NFS filesystem. This is because fcntl() file locking is broken on many NFS implementations.

I too experienced the Overseerr slowness with Overseerr running on NVME SSDs, but via a NFS mount. I tried the aforementioned DNS and MTU edits and they didn't work for me. I tried NFS v3/v4 and a ton of different NFS options before giving up on NFS. Eventually, I moved my docker volume directory from NFS to a regular filesystem and now Overseerr is very fast.

I also resolved performance issues in a couple other services that rely on SQLite with the same move off of NFS. If you use NFS to host your Overseerr data, then I strongly recommend moving to iSCSI or CIFS.

**Comment by wiffleman** [comment]
> > > Lowering the MTU to 1400 on the Docker container has resolved the slowness on my Overseerr install for about 2 months now.
> > > I had the constant slowdowns until I specifically set the MTU. I have had zero slowdowns since.
> > 
> > 
> > Curious how you did this? Wondering if you can let me know if there are docker-compose commands I can write to make this change?
> > I have also recently tried changing to the stock sctx/overseerr image as opposed to the linuxserver one. I'm not sure if it's just my conscience playing with me but it _seems_ to load quite a bit faster... will contineu to monitor.
> 
> Here's the network section of my XML (a few posts up from here)
> 
> [#3510 (comment)](https://github.com/sct/overseerr/issues/3510#issuecomment-1860896187)

Hm tried to edit this in my docker-compose but am getting varying errors. Not sure I can make this work. Strange.

**Comment by wormangel** [comment]
> 10 to 25 seconds to load most of the elements on the page:
> 
> ![image](https://private-user-images.githubusercontent.com/8156439/298628994-7562a81f-e8ce-48d6-a06a-bed98ca7d18e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY0MjEwNDMsIm5iZiI6MTcwNjQyMDc0MywicGF0aCI6Ii84MTU2NDM5LzI5ODYyODk5NC03NTYyYTgxZi1lOGNlLTQ4ZDYtYTA2YS1iZWQ5OGNhN2QxOGUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMjhUMDU0NTQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTg2MGVlYTY4NGEyZjU5YTMxNjI5YzViNTBhMTg5ZjY0NGE5ZmEyZGFjZjM0MjdlMTZhZmQwNGFmNDExNDkzNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Ex6W7FuwZo_jjHO6Jz1v-46CGZEluZaItPR2-m3ovdc)
> 
> Seems like a lot of elements are simply stalled and sat around waiting for overseerr to process the request:
> 

I'm not a frontend expert, but would that mean that this issue/bug has a _component_ of poor implementation? Meaning there's room for a PR improving e.g. the way the screen is drawn, handle requests and responses, handle parallelization, something like that? 
Something that gave me a hunch about the existence of those was a basic UX 'annoyance': when Overseerr is displaying the very slow/laggy performance, navigating to a different page (e.g. from 'users' to 'requests' or to home, etc) also lags to show any feedback, meaning the processing of the click event is likely not taking place immediatelly and cancelling whatever was loading or blocking (like quick responsive UIs _usually_ do, so the user immediatelly feels the click went through).

My installation (linuxserver's Docker, on Synology DS420+) shows the same slowness symptoms as everybody else's above, I'm yet to try some of the suggested "fixes" but I have been thinking of checking Overseer's code for performance low-hanging fruits..

**Comment by timdonovanuk** [comment]
> I'm not even running in a docker container 😄 The MTU tweak seems to have worked, but it's really random, sometimes I think I've fixed it but the issue comes back. It's certainly not a lack of resources on my server.

And it's back...so the MTU fix is just a red-herring. 

> I'm not a frontend expert, but would that mean that this issue/bug has a _component_ of poor implementation? Meaning there's room for a PR improving e.g. the way the screen is drawn, handle requests and responses, handle parallelization, something like that? 

Not entirely sure, but it seems like the frontend is blocked waiting for the service to process stuff. Whether the frontend is to blame or the backend, no idea.

**Comment by Malnes** [comment]
This is just a theory, but it seems like the database gets locked while resources are being downloaded, and the resources are loading very slowly (probably due to throttling). I tried logging into two different browsers. Once one of them gets stuck loading, the other one does not respond until the first one has finished loading. After using the site for a while, it feels faster because many of the usual movies/series have been cached.

If the connection to the SQLite database remains active and locked while waiting for assets to download, that would be quite stupid as this would block any usage for all users while an asset is loaded, but it could explain the issues and should be easily solvable. Additionally, transitioning to a more robust database system would not be a bad idea.

Keep in mind that I don't know what I'm talking about. Just a thought.

**Comment by motoridersd** [comment]
I have two Overseerr instances on different hosts, both running in docker under Ubuntu. One runs fast, but the other has always been super slow.

Reading through this issue thread, I read about the MTU setting, but the ipv6 settings jumped out at me.

I have IPV6 disabled in my network and the problem was that Overseerr was trying to use it and timing out.

The server that runs fine for me has proper IPV6 implementation.

So, if you don't use IPV6 in your network, or you're not sure, try disabling it.

I added this to my compose and the performance is on par with my properly working host.

```
networks:
  default:
    enable_ipv6: false
    driver: bridge
    driver_opts:
      com.docker.network.enable_ipv6: "false"
```

**Comment by timdonovanuk** [comment]
Disabling ipv6 did nothing and I'm not running in docker. The comment about database locking could be a possibility.

**Comment by spencerfrost** [comment]
Holy crap I'm so glad it's not just me. I've been thinking that I did something wrong this whole time.

**Comment by stephen304** [comment]
Same issue here. I'm already running with the db in a folder on the HDD not over NFS or samba, and I tried disabling ipv6 but no dice. Requests frequently take forever to complete (resulting in timeouts on my reverse proxy) but my nas is sitting at 2-3% cpu.

It would be interesting to log all DB queries and see if any query is happening a lot when loading the discover page, or to profile where time is spent on those api calls that take forever.

**Comment by ok1ok1** [comment]
> misconfigured ACL rule

What ACL rule was misconfigured, what did you change?

**Comment by gonsaaa** [comment]
I had this slowness too and for me what fixed it was removing the `network_mode: bridge` in my docker compose.
Immediatly became super fast.

**Comment by stephen304** [comment]
So I did a little tinkering and tried a few things to rule out some causes. I've tried pretty much all the solutions suggested here to no avail.

I had around 800 requests so I deleted all available requests older than 1 month and that took me down to under 100 total requests (including pending). I vacuumed the DB and checked the table sizes. Besides requests I have the media / seasons tables (forget what they're called exactly) with around 1800-2000 rows which seems reasonable. This unfortunately didn't have any impact on the slow queries.

Next I looked into the api calls that were slow, which was most of them. I noticed that once I had loaded a page or two of a specific discover or the movies page, that page loaded fast until I restarted overseerr or loaded a page of movies/tv that hadn't been loaded before - so my next thought was some kind of caching. The frontend does make a lot of /tv/:id and /movie/:id requests which seem to slow down UI loading, and I noticed that those metadata requests are basically proxied to TMDB using an in memory cache with TTLs of around a day or so. I figured maybe TMDB just throttles a lot (doesn't actually seem to be the problem) and maybe cached entries expire too soon since only me and one other person use this overseerr instance. Nevertheless I figured it was worth a shot to make a long-lived disk-backed cache that I can pre-populate with the main discover pages on a crontab to test this theory. I made a new redis cache with disk persistence to replace the node-cache only for movie/tv metadata api calls and no TTL, and after warming the cache with lots of page loading and once I saw that no new entries were being cached (and thus it should have been getting all metadata from redis), sadly the performance was just as bad.

So I think poor TMDB api performance can be ruled out. This leads me to believe my original theory of general poor sqlite performance which was my initial hunch and why I first tried to slim down the requests and VACUUM. This is also supported by the fact that queries such as /me and /status are also super slow even though they make no external api calls (* /status does make external calls to github for version checking but this is skipped if the commitTag is local it looks like). Especially interesting is that the /request/:id endpoint is also super slow, which **I believe** should just be retrieving a single row from sqlite.

Finally I added an ssd read/write cache to my btrfs (and reverted back to the :latest docker image) and everything is super fast now. My nas runs a handful of other containers that constantly cause ~50Mbit of disk usage which results in a little bit of constant disk latency - somehow overseerr is the only app that seems to care about this as most of my other apps are a little sluggish but work perfectly fine - at least not on the level of waiting literally 4-5 minutes for a page to finish loading. So my conclusion so far is that overseerr+sqlite must be very sensitive to disk latency. It would be interesting to see if switching to postgres or something else would make a difference. Since typeorm supports a range of DB types this might not be too hard - I just don't know how such a change would impact migrations and whether a different schema needs to be made for that to work. 

If anyone else with these performance issues is able to try running the sqlite DB off of a local (non-network mount) fast ssd (for reference my cache SSDs are a pair of mirrored 970 evo plus 1TB), it would be good to know if that solves anyone else's problem too. I'm still not 100% sure if I'm going to be hit by poor performance again randomly since this issue seems to be a bit flaky - but I really think low latency random access in sqlite could be the cause.

**Comment by stephen304** [comment]
@rxbrad Sadly for me explicitly defining those network parameters made no difference, and I didn't make any changes to my docker compose when adding the btrfs SSD cache. It's possible that there are multiple reasons why everyone here is experiencing slowness, and in my case the background disk read latency is probably higher than most other people (to the point where all of my selfhosted apps are a little sluggish) which I think is a major difference between you and I.

**Comment by rxbrad** [comment]
@stephen304 Yep, it continues to be a mystery what the source of the slowdown is....  As noted, my setup has been running great for quite awhile.  Will share lots of variables on my install that may or may not be involved.  I already shared my Compose XML above.

- All config files & folder visible to Oversseer under /app/config have 755 permissions
- Experimental Image Caching setting is enabled; Proxy & CSRF Protection Disabled
- Overseerr is opened to the Internet through a Cloudflare tunnel. LAN & external Cloudflared URL performance is identical.
- Plex Settings: Plex & Overseer are on the same machine, and Overseerr connects to Plex via http://192.168.*.*:32400
- Tautulli is connected.  Tautulli is running external to the Overseerr machine, offsite on Oracle Cloud (it also ran great on the same machine connected via 192.168.*.*)
- Radarr: I have one server on the same machine, connected to Overseerr via http://192.168.*.*:7878
- Sonarr: I have one server on the same machine, connected to Overseerr via http://192.168.*.*:8989
- Email notifications are the only enabled notifications
- Users only login with Plex logins (local sign-in disabled); New Plex sign-in disabled

**Comment by c11umw** [comment]
Mine's been good for a long time too since I moved it away from Synergy Docker.
Now it runs in it's own Proxmox LXC (still in docker) and is zippy fast.

**Comment by jacksoncooper12** [comment]
@stephen304 

Previously ran mine on my synology NAS, experienced the slowdown, migrated to NVME drive on a new system, seemed better for a day or two (as is the case with many of these fixes), then went right back to taking forever to load anything.

**Comment by notjosh** [comment]
👋 hiya all, I took a bit of a deep dive on this and was able to solve it - at least for my use case (and explain why many folks can't reproduce the issue!). it's _possible_ there are more reasons for slowdowns, but this might help some people.

tldr: it's DNS. specifically, it's DNS within Docker.
tldr solution: edit your `/etc/docker/daemon.json` file to have your network's DNS (or use `- dns: <ip address>` in `docker-compose.yml` etc

basically, _every request_ within Overseerr was hitting Docker's internal DNS, timing out after 4 seconds, then moving on to _actual_ DNS servers. that's why some people using bridge/host networking saw improvements, or non-Docker dev environments couldn't reproduce it.

anyway, the deep dive, if anyone wants to see if it's relevant for their setup:

<details>
<summary>[click me] reproduction steps/walkthrough</summary>
I started by identifying a single slow call (`https://overseerr.mydomain.tld/api/v1/discover/watchlist`), commented out _most_ of the code within the container, until it was pretty much only doing a single `axios.get()` (i.e. no caching, no gzip, etc) to the Plex metadata server. it was still taking ~5 seconds.

(I figured the Plex metadata API was doing some rate limiting, however running via `curl` locally was fast every time.)

I noticed the container had `wget` available, so I ran the command within the container. it still took around 5 seconds. weird!

note: at this point we know it's not an Overseerr problem, and it's why people can't reproduce it in local (non-Docker) dev environments etc.

the container has `nslookup`, so I ran it within the container:

```
# time nslookup metadata.provider.plex.tv
Server:         127.0.0.11
Address:        127.0.0.11:53

[... response data ...]

real    0m 4.00s   <- 4 seconds!
user    0m 0.00s
sys     0m 0.00s
```

and then by specifying my actual DNS server at home:

```
/app # time nslookup metadata.provider.plex.tv 192.168.x.x
Server:         192.168.x.x
Address:        192.168.x.x:53

[... response data ...]

real    0m 0.00s   <- instant
user    0m 0.00s
sys     0m 0.00s
```

(note: the slow lookup was repeatable, not just fast the 2nd time because of DNS caching)

I found a SO post ([here](https://serverfault.com/questions/1135897/docker-container-dns-slow-4-seconds-with-dns-resolver-as-container)) that described the exact 4 second slowdown, so we know it's a Docker problem (or feature, I guess).

so I updated `/etc/docker/daemon.json`, restarted Docker, and re-ran the `nslookup` steps: instant responses for both. and Overseerr runs virtually instantly now.

note: this will almost certainly break setups that rely on using container names to lookup IPs. but at least Overseerr will be fast ;)
</details>

fwiw, I found a couple of other points that might improve some Overseerr performance in specific ways, but didn't seem to impact my setup much:

- `node-cache` has a config option `useClones` which defaults to `true`, and can be expensive on large cache values. some folks have reported improvements setting that to `false`
- the newer version of `connect-typeorm` has a fix to prevent writing the `expiredAt` value on every request ([ref](https://github.com/freshgiammi-lab/connect-typeorm/issues/31)), which might unlock SQLite reads on busy servers

**Comment by Gaarindor** [comment]
Confirmed @notjosh findings. Setting DNS internal to the container resolved my painfully slow overseer issues. Thanks @notjosh !

**Comment by astubbs** [comment]
For those of you running pihole and using that for your docker dns server, like I was, and forgot this step:

https://github.com/pi-hole/docker-pi-hole/?tab=readme-ov-file#quick-start
> 3. Use the Pi-hole web UI to change the DNS settings Interface listening behavior to "Listen on all interfaces, permit all origins", if using Docker's default bridge network setting. (This can also be achieved by setting the environment variable DNSMASQ_LISTENING to all)

My pihole was pingable from containers, but all dns requests on port 53 were being ignored and timing out.

Changing this setting, I reverted back to the default DNS setup for docker and all good now.

So yeah, makes sure whatever dns server you're trying to talk to at first isn't ignoring requests.

Turns out, Overseerr is actually lightening fast :) yay!

**Comment by maluramichael** [comment]
I have the same problem. Since i put overseer behind traefik everything is slow as hell. 20 Second load times for the first /api/v1/search?page=1&query=test request.

**Comment by matt-fb** [comment]
> 👋 hiya all, I took a bit of a deep dive on this and was able to solve it - at least for my use case (and explain why many folks can't reproduce the issue!). it's _possible_ there are more reasons for slowdowns, but this might help some people.
> 
> tldr: it's DNS. specifically, it's DNS within Docker. tldr solution: edit your `/etc/docker/daemon.json` file to have your network's DNS (or use `- dns: <ip address>` in `docker-compose.yml` etc
> 
> basically, _every request_ within Overseerr was hitting Docker's internal DNS, timing out after 4 seconds, then moving on to _actual_ DNS servers. that's why some people using bridge/host networking saw improvements, or non-Docker dev environments couldn't reproduce it.

I had set my trusnas primary nameserver to the IP address of my old router after migrating to a new router.
The steps that @notjosh listed, especially the nslookup, helped me realize I had a DNS problem too.
It was causing overseerr to be incredibly slow and frankly I don't know how anything was working with that mistake.

Thanks!!!

**Comment by azaricstefan** [comment]
Any chance for proper solution here?

**Comment by ChuckyNorris** [comment]
I have this problem, seems to be because I have tailscale on the same host and MagicDNS was enabled in tailscale, disabling MagicDNS really helped.

**Comment by ShannonSG** [comment]
> * Synology ds1621+
> * Docker (was) on HDD
> 
> Tried quite a bit to solve it, including debugging the app code in the container (+DNS/network) with no obvious issues on the app side...
> 
> Decided to add an nvme pool and move docker there, on the way noticed that here I have a checkmark here: ![Control Panel > Shared Folder > Docker [edit]](https://private-user-images.githubusercontent.com/488812/278136578-c9660c66-7a66-4d6d-8b7d-3f249328c23a.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjM3ODMzMDYsIm5iZiI6MTc2Mzc4MzAwNiwicGF0aCI6Ii80ODg4MTIvMjc4MTM2NTc4LWM5NjYwYzY2LTdhNjYtNGQ2ZC04YjdkLTNmMjQ5MzI4YzIzYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTEyMlQwMzQzMjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMTY0N2UzZmMyZTEwODA1ZTRjZDc2ZjEyYmI5NTVjMzRiNjkzNjZlZjdlODBkMjA5NDM3ZDJlZjcxMTdlNjBkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.OWZWUuLREgg0qhNlQw3ySV3BmdQIN_C7nEbiHSQ2okk)
> 
> _Unchecking it_ before moving `docker` folder to the nvme pool solved the issue for me.
> 
> I suspect the "data integrity" thing was the root - Overseer is pretty aggressive with file operations (cache) and Synology with counting checksums on it :)
> 
> UPD: in theory, a simple ways to check: copy existing data folder to a different location where the "data integrity" is off and mount it.

This was absolutely the issue for me, thank you for posting. For anyone running this on Synology I would start here.

**Comment by nathang21** [comment]
> This was absolutely the issue for me, thank you for posting. For anyone running this on Synology I would start here.

In my case, I already have a `docker` shared folder on an SSD, but you can not uncheck the `data integrity` option once the shared folder is already created. 

I am running 60+ containers, and none of them have the same performance issues that Overseer has exhibited, which is surprisingly. A migration to a new folder would be non-trivial for me, so not going to pursue that. I don't doubt that unchecking it solved it for you, but I am still inclined to think that the problem is exasperated at the application level.

**Comment by 48222** [comment]
In my case, I was running on TrueNAS and made two changes which resolved it and brought Overseerr to lightning fast speeds.

1. Set up router as initial DNS resolver with 2 and 3 being google and cloudflare
2. Removed the sytem domain name, by default it was 'local' which means .local was being treated as the search domain

I knew it wasn't system resources or ISP speeds issue, as my server is overpowered and am running a gigabit ISP connection - and as always, it was DNS. It's always DNS.

**Comment by Shaotastic** [comment]
> In my case, I was running on TrueNAS and made two changes which resolved it and brought Overseerr to lightning fast speeds.
> 
> 1. Set up router as initial DNS resolver with 2 and 3 being google and cloudflare
> 2. Removed the sytem domain name, by default it was 'local' which means .local was being treated as the search domain
> 
> I knew it wasn't system resources or ISP speeds issue, as my server is overpowered and am running a gigabit ISP connection - and as always, it was DNS. It's always DNS.

I've been using the TrueNas app myself, I fixed this by enabling Host Network in the app settings, runs flawlessly now.
